test/test-debuglet.ts

```diff
@@ -75,6 +75,20 @@ function verifyBreakpointRejection(re: RegExp, body: {breakpoint: any}) {
   return status.isError && hasCorrectDescription;
 }
 
+describe('CachedPromise', function() {
+  it('CachedPromise can proceed execute when resolve is called', (done) => {
+    this.timeout(2000);
+    let cachedPromise = new CachedPromise(60 * 1000);
```

Comment:
It might be more comprehensive to let the CachedPromise go stale as part of the test.

Reply:
Done.